s3/pkg/service/object.go

```diff
@@ -355,45 +358,77 @@ func (s *s3Service) GetObject(ctx context.Context, req *pb.GetObjectInput, strea
 		return err
 	}
 
-	var finalReader io.Reader
-	finalReader = reader
+	allBytes, _ := ioutil.ReadAll(reader)
```

Comment:
Will this read all the data from the reader? For example, if the object is 2G, will this read all 2G bytes to allBytes? If it is like this, this will consume a lot of memory.

Reply:
yes, it will read all, since after the read, we need to decrypt in chunks...there is no multipart download from cloud...checked...please suggest any alternative, thanks

Reply:
There is download range for cloud storage, I think. If the size is big, then download all will use too much memory.

Reply:
i checked AWS docs, could not find any...download with range...

Reply:
https://docs.aws.amazon.com/AmazonS3/latest/dev/bucket-encryption.html, here, search range please.

Reply:
thanks, did not find...am i missing something ?
![image](https://user-images.githubusercontent.com/48434725/74301934-ad79a200-4d7a-11ea-8b4b-7dff5bfe85db.png)

Reply:
Hi @rhsakarpos , have you found the range definition now? Maybe google with key word aws s3, get range can help.

Reply:
yes, found this...correct ?
```
GET /example-object HTTP/1.1
            Host: example-bucket.s3.<Region>.amazonaws.com
            x-amz-date: Fri, 28 Jan 2011 21:32:02 GMT
            Range: bytes=0-9
            Authorization: AWS AKIAIOSFODNN7EXAMPLE:Yxg83MZaEgh3OZ3l0rLo5RTX11o=
            Sample Response with Specified Range of the Object Bytes
```